### PR TITLE
Add implementation

### PR DESCRIPTION
**Pull request by omrilotan**
Measure browser page performance timing and reduce it to a results object using the [navigation timing API](https://www.w3.org/TR/navigation-timing/)

This program collects each metrics' total time from `timeOrigin` (falls back to `navigationStart`), so metric calculations can be performed by the analyser, wherever the data is reported to.

### Basic use
```js
import { pageTiming } from 'page-timing';
const metrics = ['domInteractive', 'loadEventEnd'];

pageTiming({metrics});

console.log(...results);
```
Output
```json
[
  [
    "domInteractive",
    208.010986328125
  ],
  [
    "loadEventEnd",
    431.010986328125
  ]
]
```

### Key value pairs
```js
const now = Date.now();
const reducer = (accumulator, [key, value]) =>
  [...accumulator, `browser_performance.${page_name}.${key}:${parseInt(value)}|ms`]
;
const results = pageTiming({
    reducer,
    metrics,
});

sendMetricsToStatsD(...results);
```
Output
```json
[
  "browser_performance.homepage_loggedout.domInteractive:208|ms",
  "browser_performance.homepage_loggedout.loadEventEnd:431|ms"
]
```

### Enriched objects
```js
const reducer = (accumulator, [key, value]) => Object.assign(
    accumulator,
    {[key]: parseFloat(value.toFixed(2))}
);
const event = {
    group: "performance",
    type: "browser_performance",
    timings: {},
    page: "homepage_loggedout"
};
const results = pageTiming({
    reducer,
    accumulator: event.timings,
    metrics,
});

sendPreformanceEvent(...results);
```
Output
```json
{
  "group": "performance",
  "type": "browser_performance",
  "metrics": {
    "domInteractive": 208.01,
    "loadEventEnd": 431.01
  },
  "page": "homepage_loggedout"
}
```

### CODE REVIEW

File: src/supported/spec.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+const memoiseModule = require('../memoise');
+memoiseModule.memoise = (key, fn) => fn();
+
+describe('supported', () => {
+    let supported;
+    const performance = Object.getOwnPropertyDescriptor(window, 'performance');
+
+    beforeEach(() => {
+        delete require.cache[require.resolve('.')];
```
Comment (ealush) [inline src/supported/spec.js]: why not not after?

```
afterEach(() => {
    Object.defineProperty(window, 'performance', performance);
    delete require.cache[require.resolve('.')];
    supported = undefined;
});
```

*Update*: I know that reassigning `supported` should be enough for removing the reference to the old module, but I do not think they should depend on each other. WDYT?



--- same comment applies to other tests.

File: src/measure/index.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+/**
+ * Get the measurement diff from start
+ * @param  {String} measurement
+ * @return {Number}
+ */
+export const measure = (measurement) => {
+    const t = window.performance.timing[measurement];
+
+    return t > start() ? t - start() : 0;
```
Comment (ealush) [inline src/measure/index.js]: Can there be cases in which `t` is smaller than start?
If not, then `return t - start()` should suffice.

Comment (omrilotan) [reply src/measure/index.js]: `t` can be 0

File: src/index.js [hunk 3]
```diff
@@ -1 +1,40 @@
...
+                    }
+
+                    const value = measure(metric);
+
+                    // Ignore empty entries
+                    if (value <= 0) {
+                        return results;
+                    }
+
+
```
Comment (ealush) [inline src/index.js]: ...

File: src/index.js [hunk 4]
```diff
@@ -1 +1,40 @@
...
+import PERFORMANCE_METRICS from './performance-metrics';
+import { measure } from './measure';
+import { supported } from './supported';
+
+const DEFAULT_REDUCER = (accumulator, [key, value]) => [...accumulator, [key, value]];
+const DEFAULT_ACCUMULATOR = [];
+
+/**
+ * An opinionated reducer on performance.timing object
+ * @param  {Array}    [options.metrics]     Requested metrics
```
Comment (ealush) [inline src/index.js]: `Array.<Number>`

File: src/performance-metrics/index.json [hunk 5]
```diff
@@ -0,0 +1,23 @@
+[
```
Comment (ealush) [inline src/performance-metrics/index.json]: I know `.includes` is much prettier, but if you make this an object, you'll save iterating this array during reduce

File: src/memoise/spec.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        expect(memoise(key, () => 'Thang')).to.equal('Thang')
+    );
+
+    it('Should memoise the result', () => {
+        memoise(key, () => 'Thing');
+        memoise(key, () => 'Thang');
+        expect(memoise(key)).to.equal('Thing');
+    });
+
+    it('Should not be able to memoise by primitive', () =>
```
Comment (ealush) [inline src/memoise/spec.js]: Should return without memoising  primitives

**Pull request merged**